<?xml version="1.0"?>
<launch>
    <!-- Receives CmdVel, takeoff, reset and land messages from the MQTT and publishes on ROS -->
    <!-- <param name="mqttBroker" value="localhost" /> -->

    <node name="followLine2" pkg="create_eyes" type="followLine2.py" output="screen" clear_params="true">
        <param name="loop_rate" value="200" />

    </node>
    <!-- Sends Image and Navdata to the Mqtt broker -->
    <node name="controller" pkg="create_eyes" type="controller.py" output="screen" clear_params="true">
        <!-- Make sure the topics published are subscribed with the exact name -->
        <param name="publishedMqttTopic_navdata" value="/mqtt/navdata"/>
    </node>

    <node name="statePublisher" pkg="create_eyes" type="statePublisher.py" output="screen" clear_params="true">
        <param name="sampleParam" value="none" />
    </node>

    <node name="sonarPublisher" pkg="create_eyes" type="sonarNode.py" output="screen" clear_params="true">
        <param name="sampleParam" value="none" />
    </node> 

  <arg name="name" default="raspicam_node" />

  <!-- Raspicam Node -->

  <node type="raspicam_node" pkg="raspicam_node" name="$(arg name)" output="screen">

    <param name="camera_info_url" value="package://$(arg name)/camera_info/camera_module_v2_640x480.yaml"/>
    <param name="width" value="640"/>
    <param name="height" value="480"/>
    <param name="framerate" value="30"/>
    <param name="camera_frame_id" value="raspicam"/>

  </node>

  <node pkg="rosservice" type="rosservice" name="start_capture" args="call --wait $(arg name)/start_capture"/>

  <!-- iRobotCreate2 -->

  <arg name="id" default="0"/>
  <arg name="base_name" default="iRobot_"/>
  <arg name="port" default="/dev/ttyUSB0"/>
  <arg name="publish_name" default="true"/>

  <node ns="$(arg base_name)$(arg id)" name="irobotdriver" pkg="irobotcreate2" type="irobotcreate2" respawn="false" output="screen">
      <!-- <param name="id_" value="$(arg id)"/> -->
      <!-- <param name="base_name_" value="$(arg base_name)"/> -->
    <rosparam param="poseCovariance">[1e-2, 0,    0,    0,    0,    0,
                                      0,    1e-2, 0,    0,    0,    0,
                                      0,    0,    1, 0,    0,    0,
                                      0,    0,    0,    1, 0,    0,
                                      0,    0,    0,    0,    1, 0,
                                      0,    0,    0,    0,    0,    1e-1]</rosparam>
    <rosparam param="twistCovariance">[1e-2, 0,    0,    0,    0,    0,
                                      0,    1, 0,    0,    0,    0,
                                      0,    0,    1, 0,    0,    0,
                                      0,    0,    0,    1, 0,    0,
                                      0,    0,    0,    0,    1, 0,
                                      0,    0,    0,    0,    0,    1e-2]</rosparam>

    <param name="port_" value="$(arg port)"/>
    <param name="publish_name_" value="$(arg publish_name)"/>
    <param name="publishTf" value="true"/>
  </node>


</launch>

